packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -1,13 +1,145 @@
+import TrezorConnect, { FeeLevel, TokenInfo } from '@trezor/connect';
 import { Transaction, SystemProgram, PublicKey } from '@solana/web3.js';
 import {
     FormState,
     PrecomposedTransactionFinal,
+    ComposeActionContext,
+    ExternalOutput,
+    PrecomposedTransaction,
+    PrecomposedLevels,
 } from '@suite-common/wallet-types';
 import { notificationsActions } from '@suite-common/toast-notifications';
 import { Dispatch, GetState } from 'src/types/suite';
 import {
+    calculateMax,
+    calculateSolFee,
+    calculateTotal,
+    formatAmount,
+    getExternalComposeOutput,
     getLamportsFromSol,
 } from '@suite-common/wallet-utils';
+import BigNumber from 'bignumber.js';
+
+// TODO(vl): Implement send-max
+// TODO(vl), phase 2: Implement custom Prioritization Feetat
+
+// TODO(vl): Handle token transfers -- see ETH implementation
+const calculate = (
+    availableBalance: string,
+    output: ExternalOutput,
+    feeLevel: FeeLevel,
+    token?: TokenInfo,
+): PrecomposedTransaction => {
+    // TODO(vl): handle prioritisation fee
+    // 1, 0, 0 = discard prioritisation fee for now
+    // Default args include the prioritisation fee
+    const feeInSatoshi = calculateSolFee(1, 0, 0);
+    let amount: string;
+    let max: string | undefined;
+    if (output.type === 'send-max' || output.type === 'send-max-noaddress') {
+        max = calculateMax(availableBalance, feeInSatoshi);
+        amount = max;
```

Comment:
I would remove this and just set `max` to explicit `undefined` in the payload data.